tests/main/interfaces-password-manager-service/task.yaml

```diff
@@ -0,0 +1,56 @@
+summary: Ensure that the password-manager-service interface works
+
+systems:
+    - -ubuntu-core-16-*
+
+prepare: |
+    . $TESTSLIB/pkgdb.sh
+    echo "Ensure we have a working gnome-keyring"
+    distro_install_package gnome-keyring dbus-x11
+    snap install --edge test-snapd-password-manager-consumer
+
+restore: |
+    . $TESTSLIB/pkgdb.sh
+    distro_auto_remove_packages gnome-keyring dbus-x11
+    kill $(cat dbus-launch.pid)
+    rm -f  dbus-launch.pid
+
+execute: |
+    CONNECTED_PATTERN=":password-manager-service +test-snapd-password-manager-consumer"
+    DISCONNECTED_PATTERN="(?s).*?\n- +test-snapd-password-manager-consumer:password-manager-service"
+
+    echo "Ensure things run"
+    eval $(dbus-launch --sh-syntax)
+    eval $(printf password|gnome-keyring-daemon --login)
+    eval $(gnome-keyring-daemon --start)
+    echo "$DBUS_SESSION_BUS_PID" > dbus-launch.pid
+
+    echo "Then it is not shown as connected"
+    snap interfaces | grep -Pzq "$DISCONNECTED_PATTERN"
```

Comment:
a nit: if this fails, you'd get a much nicer error message if you were using `MATCH` instead of `grep -Pzq`.
`MATCH` is implemented via `grep -E`, so you'd have to drop the initial `(?s).*?\n` from `DISCONNECTED_PATTERN`, but I'm not sure that's buying you anything.

The same applies to the following greps also, of course.